File: Exemplos/App_eXperience_Java/Elgin_AppExperience_Java-font/app/src/main/java/com/elgin/elginexperience/Pix4Page.java
package com.elgin.elginexperience;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.Button;
import android.widget.LinearLayout;

import static com.elgin.elginexperience.Services.Pix4.Produto.*;
import static com.elgin.elginexperience.Services.Pix4.Framework.*;

import com.elgin.elginexperience.Services.Pix4.Pix4ImagesStorageService;
import com.elgin.elginexperience.Services.Pix4.Pix4Service;

public class Pix4Page extends AppCompatActivity {

    // Objeto com implementação do uso do PIX 4.
    private Pix4Service pix4Obj;

    // Objeto que contém a lógica de salvamento das imagens que serão carregadas dentro do dispositivo e, posteriormente, carregado no PIX4.
    private Pix4ImagesStorageService pix4ImagesStorageService;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_pix4_page);

        pix4ImagesStorageService = new Pix4ImagesStorageService(this);
        pix4ImagesStorageService.executeStoreImages();

        pix4Obj = new Pix4Service(this);
        pix4Obj.abreConexaoDisplay();

        initQrCodeViews();

        initProductViews();

        initActionButtons();
    }

    private void initQrCodeViews() {
        final LinearLayout buttonJava, buttonDelphi, buttonFlutter, buttonXamarinAndrod, buttonXamarinForms, buttonReactNative, buttonKotlin, buttonIonic;

        buttonJava = findViewById(R.id.buttonJava);
        buttonDelphi = findViewById(R.id.buttonDelphi);
        buttonFlutter = findViewById(R.id.buttonFlutter);
        buttonXamarinAndrod = findViewById(R.id.buttonXamarinAndroid);
        buttonXamarinForms = findViewById(R.id.buttonXamarinForms);
        buttonReactNative = findViewById(R.id.buttonReactNative);
        buttonKotlin = findViewById(R.id.buttonKotlin);
        buttonIonic = findViewById(R.id.buttonIonic);

        buttonJava.setOnClickListener(v -> pix4Obj.apresentaQrCodeLinkGihtub(JAVA));
        buttonDelphi.setOnClickListener(v -> pix4Obj.apresentaQrCodeLinkGihtub(DELPHI));
        buttonFlutter.setOnClickListener(v -> pix4Obj.apresentaQrCodeLinkGihtub(FLUTTER));
        buttonXamarinAndrod.setOnClickListener(v -> pix4Obj.apresentaQrCodeLinkGihtub(XAMARIN_ANDROID));
        buttonXamarinForms.setOnClickListener(v -> pix4Obj.apresentaQrCodeLinkGihtub(XAMARIN_FORMS));
        buttonReactNative.setOnClickListener(v -> pix4Obj.apresentaQrCodeLinkGihtub(REACT_NATIVE));
        buttonKotlin.setOnClickListener(v -> pix4Obj.apresentaQrCodeLinkGihtub(KOTLIN));
        buttonIonic.setOnClickListener(v -> pix4Obj.apresentaQrCodeLinkGihtub(IONIC));
    }

    private void initProductViews() {
        final LinearLayout buttonAbacaxi, buttonBanana, buttonChocolote, buttonDetergente, buttonErvilha, buttonFeijao, buttonGoiabada, buttonHamburguer, buttonIogurte, buttonJaca;

        buttonAbacaxi = findViewById(R.id.buttonAbacaxi);
        buttonBanana = findViewById(R.id.buttonBanana);
        buttonChocolote = findViewById(R.id.buttonChocolate);
        buttonDetergente = findViewById(R.id.buttonDetergente);
        buttonErvilha = findViewById(R.id.buttonErvilha);
        buttonFeijao = findViewById(R.id.buttonFeijao);
        buttonGoiabada = findViewById(R.id.buttonGoiabada);
        buttonHamburguer = findViewById(R.id.buttonHamburguer);
        buttonIogurte = findViewById(R.id.buttonIogurte);
        buttonJaca = findViewById(R.id.buttonJaca);

        buttonAbacaxi.setOnClickListener(v -> pix4Obj.adicionaProdutoApresenta(ABACAXI));
        buttonBanana.setOnClickListener(v -> pix4Obj.adicionaProdutoApresenta(BANANA));
        buttonChocolote.setOnClickListener(v -> pix4Obj.adicionaProdutoApresenta(CHOCOLATE));
        buttonDetergente.setOnClickListener(v -> pix4Obj.adicionaProdutoApresenta(DETERGENTE));
        buttonErvilha.setOnClickListener(v -> pix4Obj.adicionaProdutoApresenta(ERVILHA));
        buttonFeijao.setOnClickListener(v -> pix4Obj.adicionaProdutoApresenta(FEIJAO));
        buttonGoiabada.setOnClickListener(v -> pix4Obj.adicionaProdutoApresenta(GOIABADA));
        buttonHamburguer.setOnClickListener(v -> pix4Obj.adicionaProdutoApresenta(HAMBURGUER));
        buttonIogurte.setOnClickListener(v -> pix4Obj.adicionaProdutoApresenta(IOGURTE));
        buttonJaca.setOnClickListener(v -> pix4Obj.adicionaProdutoApresenta(JACA));
    }

    private void initActionButtons() {
        Button buttonShowShoppingList = findViewById(R.id.buttonShowShoppingList);
        buttonShowShoppingList.setOnClickListener(v -> pix4Obj.apresentaListaCompras());

        Button buttonLoadImages = findViewById(R.id.buttonLoadImagesOnPIX4);
        buttonLoadImages.setOnClickListener(v -> pix4Obj.carregarImagens());
    }
}

File: Exemplos/App_eXperience_Kotlin/Elgin_AppExperience_Kotlin-font/app/src/main/java/com/example/e1_kotlin_r/Pix4Page.kt
package com.example.e1_kotlin_r

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.LinearLayout
import androidx.appcompat.app.AppCompatActivity
import com.example.e1_kotlin_r.Services.Pix4.Framework
import com.example.e1_kotlin_r.Services.Pix4.Pix4ImagesStorageService
import com.example.e1_kotlin_r.Services.Pix4.Pix4Service
import com.example.e1_kotlin_r.Services.Pix4.Produto

class Pix4Page : AppCompatActivity() {

    // Objeto com implementação do uso do PIX 4.
    private var pix4Obj: Pix4Service? = null

    // Objeto que contém a lógica de salvamento das imagens que serão carregadas dentro do dispositivo e, posteriormente, carregado no PIX4.
    private var pix4ImagesStorageService: Pix4ImagesStorageService? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_pix4_page)

        pix4ImagesStorageService = Pix4ImagesStorageService(this)
        pix4ImagesStorageService!!.executeStoreImages()

        pix4Obj = Pix4Service(this)

        initQrCodeViews()

        initProductViews()

        initActionButtons()
    }

    private fun initQrCodeViews() {
        val buttonJava: LinearLayout = findViewById(R.id.buttonJava)
        val buttonDelphi: LinearLayout = findViewById(R.id.buttonDelphi)
        val buttonFlutter: LinearLayout = findViewById(R.id.buttonFlutter)
        val buttonXamarinAndroid: LinearLayout = findViewById(R.id.buttonXamarinAndroid)
        val buttonXamarinForms: LinearLayout = findViewById(R.id.buttonXamarinForms)
        val buttonReactNative: LinearLayout = findViewById(R.id.buttonReactNative)
        val buttonKotlin: LinearLayout = findViewById(R.id.buttonKotlin)
        val buttonIonic: LinearLayout = findViewById(R.id.buttonIonic)
        buttonJava.setOnClickListener {
            pix4Obj!!.apresentaQrCodeLinkGihtub(
                Framework.JAVA
            )
        }
        buttonDelphi.setOnClickListener { _: View? ->
            pix4Obj!!.apresentaQrCodeLinkGihtub(
                Framework.DELPHI
            )
        }
        buttonFlutter.setOnClickListener {
            pix4Obj!!.apresentaQrCodeLinkGihtub(
                Framework.FLUTTER
            )
        }
        buttonXamarinAndroid.setOnClickListener { _: View? ->
            pix4Obj!!.apresentaQrCodeLinkGihtub(
                Framework.XAMARIN_ANDROID
            )
        }
        buttonXamarinForms.setOnClickListener { _: View? ->
            pix4Obj!!.apresentaQrCodeLinkGihtub(
                Framework.XAMARIN_FORMS
            )
        }
        buttonReactNative.setOnClickListener { _: View? ->
            pix4Obj!!.apresentaQrCodeLinkGihtub(
                Framework.REACT_NATIVE
            )
        }
        buttonKotlin.setOnClickListener { _: View? ->
            pix4Obj!!.apresentaQrCodeLinkGihtub(
                Framework.KOTLIN
            )
        }
        buttonIonic.setOnClickListener { _: View? ->
            pix4Obj!!.apresentaQrCodeLinkGihtub(
                Framework.IONIC
            )
        }
    }

    private fun initProductViews() {
        val buttonAbacaxi: LinearLayout = findViewById(R.id.buttonAbacaxi)
        val buttonBanana: LinearLayout = findViewById(R.id.buttonBanana)
        val buttonChocolote: LinearLayout = findViewById(R.id.buttonChocolate)
        val buttonDetergente: LinearLayout = findViewById(R.id.buttonDetergente)
        val buttonErvilha: LinearLayout = findViewById(R.id.buttonErvilha)
        val buttonFeijao: LinearLayout = findViewById(R.id.buttonFeijao)
        val buttonGoiabada: LinearLayout = findViewById(R.id.buttonGoiabada)
        val buttonHamburguer: LinearLayout = findViewById(R.id.buttonHamburguer)
        val buttonIogurte: LinearLayout = findViewById(R.id.buttonIogurte)
        val buttonJaca: LinearLayout = findViewById(R.id.buttonJaca)
        buttonAbacaxi.setOnClickListener { _: View? ->
            pix4Obj!!.adicionaProdutoApresenta(
                Produto.ABACAXI
            )
        }
        buttonBanana.setOnClickListener { _: View? ->
            pix4Obj!!.adicionaProdutoApresenta(
                Produto.BANANA
            )
        }
        buttonChocolote.setOnClickListener { _: View? ->
            pix4Obj!!.adicionaProdutoApresenta(
                Produto.CHOCOLATE
            )
        }
        buttonDetergente.setOnClickListener { _: View? ->
            pix4Obj!!.adicionaProdutoApresenta(
                Produto.DETERGENTE
            )
        }
        buttonErvilha.setOnClickListener { _: View? ->
            pix4Obj!!.adicionaProdutoApresenta(
                Produto.ERVILHA
            )
        }
        buttonFeijao.setOnClickListener { _: View? ->
            pix4Obj!!.adicionaProdutoApresenta(
                Produto.FEIJAO
            )
        }
        buttonGoiabada.setOnClickListener { _: View? ->
            pix4Obj!!.adicionaProdutoApresenta(
                Produto.GOIABADA
            )
        }
        buttonHamburguer.setOnClickListener { _: View? ->
            pix4Obj!!.adicionaProdutoApresenta(
                Produto.HAMBURGUER
            )
        }
        buttonIogurte.setOnClickListener { _: View? ->
            pix4Obj!!.adicionaProdutoApresenta(
                Produto.IOGURTE
            )
        }
        buttonJaca.setOnClickListener { _: View? ->
            pix4Obj!!.adicionaProdutoApresenta(
                Produto.JACA
            )
        }
    }

    private fun initActionButtons() {
        val buttonShowShoppingList = findViewById<Button>(R.id.buttonShowShoppingList)
        buttonShowShoppingList.setOnClickListener { _: View? -> pix4Obj!!.apresentaListaCompras() }
        val buttonLoadImages = findViewById<Button>(R.id.buttonLoadImagesOnPIX4)
        buttonLoadImages.setOnClickListener { _: View? -> pix4Obj!!.carregarImagens() }
    }
}